cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -403,55 +403,76 @@ def _check_sg_rules_for_port(rule, port_to_check):
+        if all_network_interfaces:
+            response = AWSApi.instance().ec2.describe_network_interfaces(all_network_interfaces)
+            network_interfaces_data = {}
+            for network_interface in response:
+                network_interfaces_data[network_interface["NetworkInterfaceId"]] = network_interface
+            return network_interfaces_data
+        else:
+            return {}
+
+    def _validate(self, file_system_ids, head_node_subnet_id, are_all_security_groups_customized):
         try:
 
             # Check to see if there is any existing mt on the fs
-            file_system = AWSApi.instance().fsx.get_filesystem_info(file_system_id).file_system_data
+            file_systems = AWSApi.instance().fsx.get_file_systems_info(file_system_ids)
 
             vpc_id = AWSApi.instance().ec2.get_subnet_vpc(head_node_subnet_id)
 
-            # Check to see if fs is in the same VPC as the stack
-            if file_system.get("VpcId") != vpc_id:
-                self._add_failure(
-                    "Currently only support using FSx file system that is in the same VPC as the cluster. "
-                    "The file system provided is in {0}.".format(file_system.get("VpcId")),
-                    FailureLevel.ERROR,
-                )
+            network_interfaces_data = self._describe_network_interfaces(file_systems)
 
-            # If there is an existing mt in the az, need to check the inbound and outbound rules of the security groups
-            network_interface_ids = file_system.get("NetworkInterfaceIds")
-            if not network_interface_ids:
-                self._add_failure(
-                    "Unable to validate FSx security groups. The given FSx file system '{0}' doesn't have "
-                    "Elastic Network Interfaces attached to it.".format(file_system_id),
-                    FailureLevel.ERROR,
-                )
-            else:
-                network_interface_responses = AWSApi.instance().ec2.describe_network_interfaces(network_interface_ids)
-
-                fs_access = False
-                network_interfaces = [ni for ni in network_interface_responses if ni.get("VpcId") == vpc_id]
-                for network_interface in network_interfaces:
-                    # Get list of security group IDs
-                    sg_ids = [sg.get("GroupId") for sg in network_interface.get("Groups")]
-                    if _check_in_out_access(sg_ids, port=988, is_cidr_optional=are_all_security_groups_customized):
-                        fs_access = True
-                        break
-                if not fs_access:
+            # Check file systems
+            for file_system in file_systems:
+                # Check to see if fs is in the same VPC as the stack
+                file_system_data = file_system.file_system_data
+                file_system_id = file_system_data.get("FileSystemId")
```

Comment:
We shouldn't retrieve FileSystemId, VpcId and other file system data fields from here.
We have the `FsxFileSystemInfo` class that is useful to hide the internal implementation. 

I know some of them were already here but let's improve the code by adding `id`, `vpc_id`, `network_interface_ids` and other useful methods in the `FsxFileSystemInfo` class and use them here.

In this way the validator doesn't need to know the boto3 implementation and internal boto3 keys. This avoids typos and provides better layer separation.

Reply:
`file_system_data` should be a private attribute of the `FsxFileSystemInfo` class.

Reply:
Fixed